### PR TITLE
Discover diskless SBD clusters

### PR DESCRIPTION
**Pull request by arbulu89**
# Description

Discover diskless SBD clusters.
A diskless SBD consists of:
- There is not any fencing resource configured
- SBD config file must have `SBD_WATCHDOG_DEV` field
- SBD config file must not have `SBD_DEVICE` field
- SBD must no have any device configured and running

More info: https://documentation.suse.com/sle-ha/15-SP5/html/SLE-HA-all/cha-ha-storage-protect.html#sec-ha-storage-protect-diskless-sbd

## How was this tested?

Tests added

### CODE REVIEW

File: lib/trento/discovery/policies/cluster_policy.ex [hunk 1]
```diff
@@ -450,16 +452,34 @@ defmodule Trento.Discovery.Policies.ClusterPolicy do
...
       %{agent: "stonith:" <> fencing_type} ->
         fencing_type
 
       _ ->
         false
     end)
+    |> do_fencing_type(sbd)
+  end
+
+  defp do_fencing_type(nil, sbd) do
```
Comment (CDimonaco) [inline lib/trento/discovery/policies/cluster_policy.ex:467]: Can we rename it to `extract_fencing_type`?

File: lib/trento/discovery/policies/cluster_policy.ex [hunk 2]
```diff
@@ -450,16 +452,34 @@ defmodule Trento.Discovery.Policies.ClusterPolicy do
...
+      @diskless_sbd_fencing_type
+    else
+      @unknown_fencing_type
+    end
   end
 
+  defp do_fencing_type(fencing_type, _), do: fencing_type
+
+  defp diskless_sbd?(%{config: %{"sbd_watchdog_dev" => _} = config, devices: []}) do
+    not Map.has_key?(config, "sbd_device")
```
Comment (CDimonaco) [inline lib/trento/discovery/policies/cluster_policy.ex:478]: Could we write that condition with pattern matching?

Like config has the key `sbd_device` and the other clause without the key

Comment (CDimonaco) [reply lib/trento/discovery/policies/cluster_policy.ex:478]: We could also has one function that extracts the fencing type and retuurn diskless fencing type or unknown, we can skip the if at line 469

**Review by CDimonaco — COMMENTED** [review]
Overall LGTM,

I left some comments

File: lib/trento/discovery/policies/cluster_policy.ex (hunk 2 above)
Comment (arbulu89) [reply lib/trento/discovery/policies/cluster_policy.ex]: done. initially i didn't want to put the `unknown` case first, as i wanted to put the positive cases first, but yes, eventually the code is now a bit cleaner

**Review by CDimonaco — APPROVED** [review]
Great! LGTM

**Pull request merged**